### ISSUE TITLE
Create a way to mount a custom volume

### ISSUE THREAD

**Issue by leszko** [hazelcast/charts#18]
Currently, to mount a volume, which contains  custom JARs, keystore/trustore, a user needs to modify templates/*. Allow doing it just by modifying `values.yaml`.

**Comment by mesutcelik** [comment]
initContainers can be used to implement this.
https://kubernetes.io/docs/concepts/workloads/pods/init-containers/

Currently the manual upload will be done with following steps.

**Create PVC**
```
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: jar-pvc
spec:
  accessModes:
    - ReadWriteOnce
  volumeMode: Filesystem
  resources:
    requests:
      storage: 8Gi
```

**Create a Pod with the volume attached to upload Jar file**

```
apiVersion: v1
kind: Pod
metadata:
  name: jar-pod
spec:
  volumes:
    - name: custom-storage
      persistentVolumeClaim:
        claimName: jar-pvc
  containers:
    - name: app
      image: ubuntu
      volumeMounts:
        - name: custom-storage
          mountPath: /data/custom
      args: ["-c", "while true; do date >> /data/custom/dates.txt; sleep 5; done"]
      command: ["/bin/bash", "-ec", "while :; do echo '.'; sleep 5 ; done"]
```

**Upload the jar file**
```
kubectl cp my-custom.jar jar-pod:/data/custom
```
**Update helm chart(statefulset.yaml) with the following environment variable**
```
        - name: CLASSPATH
          value: "/data/custom/*"
```
**Update values.yaml** 

```
customVolume:
  persistentVolumeClaim:
    claimName: job-pvc
```
Once you completed the steps, new helm install operation will start with the new jar.

**Comment by cristina-sirbu** [comment]
Can I take this?

**Comment by mesutcelik** [comment]
Sure. I suggest you to have a look this similar feature before implementing initContainers?
https://github.com/hazelcast/charts/pull/121

**Comment by leszko** [comment]
I actually think this GH Issue is already implemented in #121 . Is there anything more we should do? CC: @mesutcelik

**Comment by mesutcelik** [comment]
I would normally solve this issue by introducing initContainers. 
https://kubernetes.io/docs/concepts/workloads/pods/init-containers/#init-containers-in-use

It is another way extending classpath without using `customVolume` but you are right #121 has already solved this issue.

@cristina-sirbu You can create another issue for `initContainers` support and implement if you want to proceed.
